listener/kcp/workspace_config.go

```diff
@@ -30,10 +42,20 @@ func virtualWorkspaceConfigFromCfg(ctx context.Context, appCfg config.Config, re
 		Name:      appCfg.ApiExportName,
 	}
 	if err := clt.Get(ctx, key, &apiExport); err != nil {
-		if errors.Is(ctx.Err(), context.DeadlineExceeded) {
-			return nil, errors.Join(ErrTimeoutFetchingAPIExport, err)
+		// if this is not a local development, we must have kubernetes.graphql.gateway apiexport
+		if !appCfg.LocalDevelopment {
+			if errors.Is(ctx.Err(), context.DeadlineExceeded) {
+				return nil, errors.Join(ErrTimeoutFetchingAPIExport, err)
+			}
+			return nil, errors.Join(ErrFailedToGetAPIExport, err)
 		}
-		return nil, errors.Join(ErrFailedToGetAPIExport, err)
+
+		// otherwise fallback to the default APIExport, but live ApiBinding watching will not work
+		if err = clt.Get(ctx, client.ObjectKey{Name: kcptenancy.SchemeGroupVersion.Group}, &apiExport); err != nil {
+			return nil, errors.Join(ErrFailedToGetAPIExport, err)
+		}
+
+		log.Warn().Msg(fmt.Sprintf("failed to find %s ApiExport, listener will not watch ApiBinding changes in realtime", appCfg.ApiExportName))
```

Comment:
Any reason to not use structured logging here?
```suggestion
		log.Warn().Str("apiexport", appCfg.ApiExportName).Msg("failed to find ApiExport, listener will not watch ApiBinding changes in realtime"))
```

Reply:
I will add it a new PR